substrate/client/rpc-spec-v2/src/sudo_session_keys/mod.rs

```diff
@@ -0,0 +1,53 @@
+// This file is part of Substrate.
+
+// Copyright (C) Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: GPL-3.0-or-later WITH Classpath-exception-2.0
+
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+//! Substrate sudo sessions key API.
+
+use serde::{Deserialize, Serialize};
+
+#[cfg(test)]
+mod tests;
+
+pub mod api;
+pub mod sudo_session_keys;
+
+/// The result of an RPC method.
+#[derive(Debug, Deserialize, Serialize, PartialEq)]
+#[serde(untagged)]
+pub enum MethodResult {
```

Comment:
maybe you could remove some boiler plate by adding to functions for success and error to this type:
-  `fn error(err: impl Into<String>) -> MethodResult`
- `fn success(result: impl Into<String>) -> MethodResult`

In order to avoid having to do `Ok(MethodResult::Err(MethodResultErr { error: api_err.to_string() }))` and just do `Ok(MethodResult::error(api_err))`

Reply:
It might also be possible to remove this wrapper type completely and just use the inner types.

Your choice :)

Reply:
I've used the `crate::MethodResult` instead to align with the spec, I think I've applied this suggestion to some other pr (probably archive), thanks! 👍